staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -2142,15 +2157,19 @@ func RunTestCount(ctx context.Context, t *testing.T, store storage.Interface) {
 		obj := &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("foo-%d", i)}}
 
 		key := fmt.Sprintf("%s/%d", resourceA, i)
-		TestPropagateStoreWithKey(ctx, t, store, key, obj)
```

Comment:
Why do we want to change the semantic here? Channeling Jordan's conservative approach, even if the update flow in propagateStore is not needed, keeping it can't hurt, right?

Reply:
What I wanted to achieve is to get rid of "TestPropagateStoreWithKey" method - to really avoid confusion about the name...

Reply:
Were getting rid of it to ensure that the key used in the store function calls is computed from the object (and thereby not suprising, etc), or something else?

Reply:
https://github.com/kubernetes/kubernetes/pull/113645 is hopefully addressing all of that for store_tests.go
I will do a similar thing later for watcher tests too.